### ISSUE TITLE
断点继续训练如何设置

### ISSUE THREAD

**Issue by elesun2018** [THUDM/CogVLM#430]
请问CogVLM大模型断点继续训练如何设置配置

![image](https://github.com/THUDM/CogVLM/assets/37207093/9dc9dea8-5e2a-4175-838b-c3b635bde52f)

![image](https://github.com/THUDM/CogVLM/assets/37207093/37c0702f-8e71-4263-a3bd-99ebfcf288ca)

args.load
![image](https://github.com/THUDM/CogVLM/assets/37207093/732e9a4e-f943-4f92-87d3-462299ebec2f)
谢谢

**Comment by elesun2018** [comment]
![image](https://github.com/THUDM/CogVLM/assets/37207093/d2543bbe-110c-4766-aa33-817fe626f5db)
通过sat如何实现
"--load", "True",

**Comment by elesun2018** [comment]
需要将"--mode", "pretrain", // finetune
出现错误Unable to load optimizer from checkpoint /home/ssd1/sfz/models/satcog1206/cogvlm-base-490/1/mp_rank_00_model_states.pt, exiting. Specify --no-load-rng or --finetune to prevent attempting to load the random state.

**Comment by elesun2018** [comment]
能否指点一下，谢谢

**Comment by elesun2018** [comment]
bash设置
 "--mode", "pretrain", 
"--load", "/home//examples/checkpoints/cogvlm-experiment-04-10-07-22",
"--from_pretrained", "/home//examples/checkpoints/cogvlm-experiment-04-10-07-22"
请问load和from_pretrained模型路径有什么区别
现在出现错误：

**Comment by elesun2018** [comment]
bash设置
 "--mode", "pretrain", 
"--load", "/home//examples/checkpoints/cogvlm-experiment-04-10-07-22",
"--from_pretrained", "/home//examples/checkpoints/cogvlm-experiment-04-10-07-22"
请问load和from_pretrained模型路径有什么区别
现在出现错误：

**Comment by elesun2018** [comment]
![image](https://github.com/THUDM/CogVLM/assets/37207093/bf8ceac3-2c5b-4296-9a47-71cfb977c318)

**Comment by elesun2018** [comment]
![image](https://github.com/THUDM/CogVLM/assets/37207093/a33cbf94-ee25-41d7-9624-960281d28063)

**Comment by elesun2018** [comment]
断点续训应该达到以下要求：加载已训练模型；继承原优化器学习率；继承原epoch；训练结果loss复现一致；加载之前的随机状态继续训练数据（已训练过的图片不再重新进行学习）
目前只能实现：加载已训练模型；继承原epoch

加入参数--no-load-rng才能继续训练，因为--from_pretrained里面不包含随机状态值。
屏蔽参数--no-load-rng，导致报错 加载-from_pretrained中的rng  
 "--from_pretrained", "satcog1206/cogvlm-grounding-generalist",

**Comment by elesun2018** [comment]
请问如何解决--from_pretrained  satcog1206/cogvlm-grounding-generalist中不包含random_rng_state的问题。
断点继续训练，为啥要加载两次模型参数，第一次加载--from_pretrained  satcog1206/cogvlm-grounding-generalis
第二次加载--load", "/home//examples/checkpoints/cogvlm-experiment-04-10-07-22",
第一次加载由于以下代码报错：
![image](https://github.com/THUDM/CogVLM/assets/37207093/9e026cf4-ba67-4342-b881-5cc9469bf520)

**Comment by elesun2018** [comment]
能否指点一下，谢谢

**Comment by elesun2018** [comment]
请问修改"--seed"值，数据迭代器顺序并没有发生变化，啥原因。谢谢

**Comment by elesun2018** [comment]
cog数据获取（findall），dataloader（next）里面都有没有指定random。

**Comment by elesun2018** [comment]
debug-CogVLM/finetune_demo/finetune_cogvlm_demo.py
请问data_b = broadcast_auto(data)的作用是什么

**Comment by elesun2018** [comment]
cog数据代码里面没有数据次序读取逻辑，断点后继续训练又从开始学习原来学过的数据样本。
最好是做成存储数据次序随机状态，断点后继承训练样本次序

**Comment by elesun2018** [comment]
空了能否解答一下，谢谢
